contrib/init/upstart/docker.conf

```diff
@@ -70,3 +70,10 @@ post-start script
 		echo "$DOCKER_SOCKET is up"
 	fi
 end script
+
+pre-stop script
+        SHUTDOWN_RUNLEVELS='06'
+        if runlevel | cut -d' ' -f2 | grep -q [$SHUTDOWN_RUNLEVELS]; then
+                docker stop -t 30 $(docker ps -q)
```

Comment:
That should already be in the container metadata via `--stop-timeout`, right?  So we really shouldn't specify anything here and just let the daemon/container default DTRT.